tests/Unit/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Test_TOV.cpp

```diff
@@ -0,0 +1,95 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cmath>
+#include <cstddef>
+#include <memory>
+#include <pup.h>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/TOV.hpp"
+#include "PointwiseFunctions/EquationsOfState/EquationOfState.hpp"  // IWYU pragma: keep
+#include "Utilities/ConstantExpressions.hpp"
+
+// IWYU pragma: no_forward_declare EquationsOfState::EquationOfState
+
+namespace {
+
+constexpr double polytropic_constant = 8.0;
+double expected_newtonian_radius() noexcept {
+  /* Analytic solution in Newtonian limit for polytropic TOV integration has
+  $R = \pi / \alpha$ ; $M = 4\pi^{2}\rho_{0c} / \alpha^{3}$
+   Here $\rho_{0c}$ is the central mass density and $\alpha = sqrt(2\pi/K)$
+   with $K$ the polytropic constant
+  */
+  return M_PI / sqrt(2.0 * M_PI / polytropic_constant);
+}
+double expected_newtonian_mass(const double central_mass_density) noexcept {
+  return 4.0 * central_mass_density * square(M_PI) /
+         (cube(sqrt(2.0 * M_PI / polytropic_constant)));
+}
+void test_tov(const std::unique_ptr<EquationsOfState::EquationOfState<true, 1>>&
+                  equation_of_state,
+              const double central_mass_density,
+              const double final_log_enthalpy,
+              const bool newtonian_limit) noexcept {
+  Approx custom_approx = Approx::custom().epsilon(1.0e-08).scale(1.0);
+  const double surface_log_enthalpy = 0.000;
+  tov::InterpolationOutput tov_out_full(equation_of_state, central_mass_density,
+                                        surface_log_enthalpy);
+  tov::InterpolationOutput tov_out_test(equation_of_state, central_mass_density,
+                                        final_log_enthalpy);
+  if (newtonian_limit) {
+    const double final_radius{tov_out_full.outer_radius()};
+    const double final_mass{tov_out_full.mass(expected_newtonian_radius())};
+    CHECK(expected_newtonian_radius() == custom_approx(final_radius));
+    CHECK(expected_newtonian_mass(central_mass_density) ==
+          custom_approx(final_mass));
```

Comment:
Do you ever check non-Newtonian limit? I'm pretty sure we discussed this before.

Reply:
I check that the interpolation works in the non-Newtonian limit (comparing the result where some interior point r is the endpoint of integration, and where the surface is the endpoint but I interpolate to r). I don't know of any analytical solutions to compare to in the relativistic limit, however.

Reply:
Also, I have checked convergence in the relativistic limit against the TOV solver in SpEC. I can send you some plots if you'd like. I did this about a month ago and discussed with Saul, I got the impression that convergence with SpEC was good enough to prove that the code works in the relativistic limit.

Reply:
Okay, well you should have a test in the code that checks the values at the surface so that we know if there is a regression :) No need to check in the bulk since the interpolation doesn't care about whether it's Newtonian or Relativistic, but I do want to make sure that if something gets changed in the solver that breaks only the relativistic case that we will have a test fail